test/cli_test.js

```diff
@@ -236,6 +239,28 @@ describe('cli', function() {
           chai.expect(spy).to.be.calledWith(sinon.match.has('env', 'production'))
         );
       });
+
+      it('leaves BROCCOLI_ENV unaltered if --environment is not set', function() {
+        const spy = sinon.spy(loadBrocfile());
+        sinon.stub(broccoli, 'loadBrocfile').value(() => spy);
+        process.env.BROCCOLI_ENV = 'production';
+
+        return cli(['node', 'broccoli', 'build', 'dist']).then(() => {
+          chai.expect(spy).to.be.calledWith(sinon.match.has('env', 'development'));
```

Comment:
This seems very very bizarre to me.  

```
BROCCOLI_ENV=production broccoli build
```

^ should _absolutely_ use production, no?

Reply:
Well this is what I am saying. Should `BROCCOLI_ENV` flow through to `options.env`, I don't know if it should, but I agree this is bizarre.
Either we allow `BROCCOLI_ENV` to proxy through to `options.env` and thus keep the values in-sync, or we say that it's a one way operation, that `--environment` will set `BROCCOLI_ENV` but `BROCCOLI_ENV` will NOT set `--environment/options.env`